usr/src/uts/common/fs/zfs/dbuf.c

```diff
@@ -2505,6 +2510,18 @@ dbuf_prefetch(dnode_t *dn, int64_t level, uint64_t blkid, zio_priority_t prio,
 	dpa->dpa_zio = pio;
 
 	/*
+	 * Minimum object to check for ARC flags
+	 */
+	db = kmem_cache_alloc(dbuf_cache, KM_SLEEP);
```

Comment:
I don't think that "faking up" the dbuf is a good idea.  Maybe we can write `_IMPL()` versions of `DBUF_IS_L2CACHEABLE` and `DBUF_IS_L2COMPESSIBLE` that take the underlying data that they need (objset, level, dnode)?